app/main.py

```diff
@@ -1,13 +1,16 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"], "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"], "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(mistake) for mistake in errors],
+            "path": file_path, "status": ("failed" if errors else "passed")}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [{"errors": [format_linter_error(fault)
+            for fault in linter_report[key]], "path": key,
+             "status": ("failed" if linter_report[key] else "passed")}
+            for (key, value) in linter_report.items()]
```

Comment:
You should use function `format_single_linter_file` to avoid code duplication

Reply:
In function #3 input function format_sigle_linter_file